lib/msf/core/payload/python/reverse_http.rb

```diff
@@ -15,6 +15,7 @@ def initialize(info = {})
       Msf::Opt::http_header_options +
       Msf::Opt::http_proxy_options
     )
+    deregister_options('HttpProxyType')
```

Comment:
Was this meant to also impact `python/meterpreter_reverse_http`? 👀 

That payload seems to include this base mixin:

https://github.com/rapid7/metasploit-framework/blob/773c6be61b311c87a09397ca64789b6827394c75/modules/payloads/singles/python/meterpreter_reverse_http.rb#L13

but it also re-registers the same options again later:
https://github.com/rapid7/metasploit-framework/blob/773c6be61b311c87a09397ca64789b6827394c75/modules/payloads/singles/python/meterpreter_reverse_http.rb#L28-L31

Which makes the option appear again:
```
msf6 payload(python/meterpreter_reverse_http) > advanced
   ... etc ...
   HttpProxyType                HTTP                                                         yes       The type of HTTP proxy (Accepted: HTTP, SOCKS)
```

Reply:
Bit of a drive-by-comment, I'll likely look at this with fresh eyes tomorrow and see what I've missed 😄